ocean_lib/ocean/ocean_assets.py

```diff
@@ -602,9 +588,11 @@ def pay_for_compute_service(
             self._start_or_reuse_order_based_on_initialize_response(
                 datasets[i],
                 item,
-                consume_market_order_fee_address,
-                datasets[i].consume_market_order_fee_token,
-                datasets[i].consume_market_order_fee_amount,
+                FeeTokenArguments(
```

Comment:
I think that the caller of this methods could construct the fees object, no?

Then it's much simpler within this method. 

Similar for other methods. The fee object should be set at a higher level, and then passed as args to lower level smart contract calls.

Reply:
here we are iterating over the response from provider, datasets + algorithm, so we don't know from the caller, beforehand, which tokens and amounts correspond. We only know the fee address, which is why we create the FeeTokenInfo here, where we have the knowledge.